eox_tagging/models.py

```diff
@@ -0,0 +1,159 @@
+"""
+Model to store tags in the database
+"""
+import logging
+import uuid
+
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models.query import QuerySet
+from django.utils import timezone
+from django.utils.encoding import python_2_unicode_compatible
+
+from eox_tagging.constants import AccessLevel, GenericType, Status
+from eox_tagging.validators import Validators
+
+log = logging.getLogger(__name__)
+
+
+class TagQuerySet(QuerySet):
+    """ Tag queryset used as manager
+    """
+
+    def valid(self):
+        """
+        Returns all valid tags
+        """
+        return self.filter(invalidated_at=None)
+
+    def invalid(self):
+        """
+        Returns all invalid tags
+        """
+        return self.exclude(invalidated_at=None)
+
+    def find_by_owner(self, owner):
+        """
+        Returns all valid tags owned by owner_id
+        """
+        model = owner.__class__
+        ctype = ContentType.objects.get_for_model(model)
+
+        return self.valid().filter(belongs_to_type=ctype,
+                                   belongs_to_object_id=owner.id)
+
+    def find_all_tags_for(self, tagged_object):
+        """
+        Returns all valid on an object
+        """
+        model = tagged_object.__class__
+        ctype = ContentType.objects.get_for_model(model)
+
+        return self.valid().filter(tagged_type=ctype,
+                                   tagged_object_id=tagged_object.id)
+
+
+@python_2_unicode_compatible
+class Tag(models.Model):
+    """Model class for tags
+       Overrides save method to validate data entries before saving
+       Also, overrides delete so softDeletion is available
+
+    Attributes:
+        tag_value: unicode value of the tag tag
+        tag_type: type of the tag
+        access: access level of the tag
+        activation_date: date to activate the tag
+        expiration-date: date to deactivate de tag
+        tagged_object: object tagged
+        belongs_to: object to which the tag belongs
+        status: status of the tag, valid or invalid
+        invalidated_at: date when the tag is soft deleted
+    """
+
+    id = models.AutoField(primary_key=True,)
```

Comment:
This is the default value, I reckon it is not necessary

Reply:
comments ?

Reply:
You're right! Sorry for the delay!